src/Native/System.Security.Cryptography.Native/CMakeLists.txt

```diff
@@ -65,7 +65,7 @@ if (APPLE)
     add_custom_command(TARGET System.Security.Cryptography.Native POST_BUILD
         COMMAND ${CMAKE_INSTALL_NAME_TOOL} -change /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib @rpath/libcrypto.1.0.0.dylib $<TARGET_FILE:System.Security.Cryptography.Native>
         COMMAND ${CMAKE_INSTALL_NAME_TOOL} -change /usr/local/opt/openssl/lib/libssl.1.0.0.dylib @rpath/libssl.1.0.0.dylib $<TARGET_FILE:System.Security.Cryptography.Native>
-        COMMAND ${CMAKE_INSTALL_NAME_TOOL} -add_rpath @loader_path $<TARGET_FILE:System.Security.Cryptography.Native>
+        COMMAND ${CMAKE_INSTALL_NAME_TOOL} -add_rpath @loader_path -add_rpath /usr/local/share/dotnet/redist $<TARGET_FILE:System.Security.Cryptography.Native>
```

Comment:
I've verified that the order passed in to this command line is the order probed.  Here's what I did:

$install_name_tool -delete_rpath @loader_path S.S.C.N.OpenSsl.dylib
$otool -l       (verify there are no LC_RPATH commands)
$cp `libcrypto.1.0.0.dylib` to both `/usr/local/share/dotnet/redist` and `/usr/local/share/dotnet/shared/Microsoft.NETCore.App/1.1.1`

$sudo install_name_tool -add_rpath /usr/local/share/dotnet/redist -add_rpath @loader_path /usr/local/share/dotnet/shared/Microsoft.NETCore.App/1.1.1/System.Security.Cryptography.Native.OpenSsl.dylib

$lldb dotnet restore
(lldb) run
(lldb) image list
  (Verify that `libcrypto.1.0.0.dylib` was loaded from `/usr/local/share/dotnet/redist` the first in the list)

And then I started over, deleting the rpaths, verifying there were no LC_RPATH commands in otool. Then I ran install_name_tool flipping the `-add_rpath` orders so `@loader_path` came first.  lldb image list showed that libcrypto was loaded next to S.S.C.N.OpenSsl.dylib.